src/algebra/module/localized_module.lean

```diff
@@ -965,6 +971,215 @@ begin
+by rw [submodule.localize_eq_span, submodule.map_span, submodule.span_span_of_tower]
+
+@[simp]
+lemma submodule.localize_bot : (⊥ : submodule R M).localize S = ⊥ :=
+by rw [← submodule.span_empty, submodule.localize_span, set.image_empty, submodule.span_empty]
+
+@[simp]
+lemma submodule.localize_top : (⊤ : submodule R M).localize S = ⊤ :=
+begin
+  rw eq_top_iff,
+  rintros x -,
+  induction x using localized_module.induction_on,
+  exact submodule.mem_localize_of_mem trivial _
+end
+
+end submodule
+
+section base_change
+
+variables (R' : Type*) [comm_ring R'] [algebra R R'] [is_localization S R']
+variables [module R' M'] [is_scalar_tower R R' M'] [module R' M''] [is_scalar_tower R R' M'']
+
+
+/-- If `M''` is an `S⁻¹R` module, then any `M →ₗ[R] M''` can be lifted to `S⁻¹M →ₗ[S⁻¹R] M''`. -/
+noncomputable
+def is_localized_module.base_change [is_localized_module S f] (g : M →ₗ[R] M'') : M' →ₗ[R'] M'' :=
+begin
+  have : ∀ (x : S), is_unit (algebra_map R (module.End R M'') ↑x),
+  { intro x,
+    have := (is_localization.map_units R' x).map (algebra_map R' (module.End R' M'')),
+    rw module.End_is_unit_iff at this ⊢,
+    convert this using 1,
+    ext n,
+    exact (algebra_map_smul _ _ _).symm },
+  refine { map_smul' := _, ..is_localized_module.lift S f g this },
+  { intros r x,
+    obtain ⟨⟨r', s⟩, e⟩ := is_localization.surj S r,
+    obtain ⟨⟨x, t⟩, rfl⟩ := is_localized_module.mk'_surjective S f x,
+    simp_rw module.End_is_unit_iff at this,
+    apply (this s).injective,
+    dsimp at e ⊢,
+    rw [mul_comm, ← algebra.smul_def] at e,
+    rw [← smul_assoc, e, algebra_map_smul, ← linear_map.map_smul, ← linear_map.map_smul,
+      ← smul_assoc, e, algebra_map_smul] }
+end
+
+@[simp]
+lemma is_localized_module.base_change_mk' [is_localized_module S f] (g : M →ₗ[R] M'') (m) (s : S) :
+    is_localized_module.base_change S f R' g (is_localized_module.mk' f m s) =
+      (is_localization.mk' R' 1 s) • (g m) :=
+begin
+  dsimp [is_localized_module.base_change],
+  rw [is_localized_module.lift_mk', module.End_algebra_map_is_unit_inv_apply_eq_iff,
+    ← smul_assoc, is_localization.smul_mk', is_localization.mk'_mul_cancel_left, map_one, one_smul],
+end
+
+variable (M'')
+
+/-- `is_localized_module.base_change` as an `R`-linear map. -/
+@[simps]
+noncomputable
+def is_localized_module.base_change_linear_map [is_localized_module S f] :
+  (M →ₗ[R] M'') →ₗ[R] M' →ₗ[R'] M'' :=
+{ to_fun := is_localized_module.base_change S f R',
+  map_add' := begin
+    intros g₁ g₂,
+    ext x,
+    obtain ⟨⟨x, t⟩, rfl⟩ := is_localized_module.mk'_surjective S f x,
+    dsimp,
+    simp only [is_localized_module.base_change_mk', linear_map.add_apply, smul_add],
+  end,
+  map_smul' := begin
+    intros r g,
+    ext x,
+    obtain ⟨⟨x, t⟩, rfl⟩ := is_localized_module.mk'_surjective S f x,
+    dsimp,
+    simp only [is_localized_module.base_change_mk', linear_map.smul_apply],
+    rw smul_comm,
+  end }
```

Comment:
I'd extract these fields as separate `@[simp]` lemmas on `is_localized_module.base_change`, so that `simp`ing `base_change_linear_map` to `base_change` won't lose the applicability of `map_add` and `map_smul`.